File: andromeda/src/models/disciplineTitleModels.ts
import { Validation, GetOptions } from "./commonModels";

export interface DisciplineTitle {
    id?: number;
    departmentId?: number;
    shortname?: string;
    name: string;
    pinned?: boolean;
}

export namespace DisciplineTitle {
    export const initial: DisciplineTitle = { name: '' };
}

export interface DisciplineTitleValidation extends Validation {
    shortnameError?: string;
    nameError?: string;
}

export namespace DisciplineTitleValidation {
    export const initial: DisciplineTitleValidation = Validation.initial;
}

export interface DisciplineTitleGetOptions extends GetOptions {
    departmentId?: number;
    departmentsIds?: number[];
    notPinned?: boolean;
}

File: Andromeda.Data/Models/DisciplineTitleModels.cs
using System.Collections.Generic;
using Andromeda.Data.Enumerations;

namespace Andromeda.Data.Models
{
    public class DisciplineTitle
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Shortname { get; set; }
        public bool Pinned { get; set; }
        public int DepartmentId { get; set; }

        public ProjectType? PinnedProjectType { get; set; }
        public List<ProjectType> NotPinnedProjectTypes { get; set; }
    }

    public class DisciplineTitleGetOptions : BaseGetOptions
    {
        public int? DepartmentId { get; set; }
        public IReadOnlyList<int> DepartmentIds { get; set; }
        public IReadOnlyList<string> Titles { get; set; }
        public bool? NotPinned { get; set; }
    }
}